### PR TITLE
Ignore CUCUMBER_PUBLISH_TOKEN during cucumber tests

### PR DESCRIPTION
**Pull request by aurelien-reeves**
Re-set it properly after each tests to be able to actually publish our
own reports.

**Is your pull request related to a problem? Please describe.**

This is an attempt to fix our own GHA

**Describe the solution you have implemented**

Unset CUCUMBER_PUBLISH_TOKEN before each cucumber tests to avoid interference within the test.
Then reset it properly after each tests to be able to publish our own reports.

### CODE REVIEW

File: features/lib/support/env.rb [hunk 1]
```diff
@@ -22,9 +22,16 @@
 end
 
 Around do |_, block|
+  original_publish_token = ENV['CUCUMBER_PUBLISH_TOKEN']
+  ENV.delete('CUCUMBER_PUBLISH_TOKEN')
```
Comment (olleolleolle) [inline features/lib/support/env.rb:26]: The value deleted from the hash is returned, so you can assign it:

```ruby
  original_publish_token = ENV.delete('CUCUMBER_PUBLISH_TOKEN')
```

**Review by olleolleolle — COMMENTED** [review]
Just poking around

**Review by luke-hill — COMMENTED** [review]
Seems fine (olle offered a slight optimisation).

One thing (Possibly one for @vincent-psarga), we mentioned a while back that `Around` hooks were likely to go. Is it potentially worth whilst we're here, popping these in the `Before` / `After`

I don't know if we're still looking to get rid of Around hooks. But I know we were around 3.x

File: features/lib/support/env.rb (hunk 1 above)
Comment (aurelien-reeves) [reply features/lib/support/env.rb]: done ✅

**Comment by aurelien-reeves** [comment]
> One thing (Possibly one for @vincent-psarga), we mentioned a while back that `Around` hooks were likely to go. Is it potentially worth whilst we're here, popping these in the `Before` / `After`
> 
> I don't know if we're still looking to get rid of Around hooks. But I know we were around 3.x

Ok, just let me know

**Review by romaingweb — APPROVED** [review]

**Comment by romaingweb** [comment]
Would it make sense to do those changes (from `Around` to `Before` / `After`) later to not block this fix?

**Comment by luke-hill** [comment]
Oki go nuts. Vincent can catch-up later if needs be.

**Pull request merged**